transformer_engine/jax/csrc/modules.cpp

```diff
@@ -980,12 +968,11 @@ void RMSNormBackward(cudaStream_t stream, void **buffers, const char *opaque, si
 
 void Quantize(cudaStream_t stream, void **buffers, const char *opaque, size_t opaque_len) {
     auto *input = buffers[0];
-    auto *amax = reinterpret_cast<float *>(buffers[1]);
+    // auto *amax = reinterpret_cast<float *>(buffers[1]);  // Bound to amax_out in primitive
```

Comment:
Hi @denera, could you help tell me what was the warning message before this change?

Reply:
It was an unused variable warning.

Apparently the compiler doesn't think `assert(amax == amax_out)` counts as using the variable.

Reply:
Thank you for pointing this out. `assert` statements are automatically removed when compiled in “Release” mode. It appears our previous build wasn’t set to “Release” mode, which is why we didn’t encounter this warning earlier.

I believe we should retain this check to safeguard against potential modifications in `cpp_extensions.py` that could break the amax mappings. I’ve noticed that using `NVTE_CHECK` doesn’t trigger the warnings, so it seems like a suitable replacement. In this PR, we can comment out the current `assert` like you did in this PR and I can create another PR to replace the `assert` with `NVTE_CHECK` as well as other assert usages.